clippy_lints/src/methods.rs

```diff
@@ -741,15 +741,18 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
             if let Some(first_arg) = iter_input_pats(&sig.decl, cx.tcx.hir.body(id)).next();
             if let hir::ItemImpl(_, _, _, _, None, ref self_ty, _) = item.node;
             then {
+                if implitem.vis == hir::Visibility::Public ||
```

Comment:
`T::add`/`T::drop` are actually private since `T` is private. So, can you use `cx.access_levels.is_exported(implitem.id)` instead of `vis`, and make `struct T` public?

Reply:
Just to understand: this change is because if I make a struct SomeStruct with an impl with public functions, those functions aren't actually public because they're called with knowledge of the struct. So then there's no need to lint them. 

Will `cx.access_levels.is_exported(implitem.id)` respect both the pub/pub-restricted keywords on the function and type?

Reply:
It returns true only for pub (or publicly re-exported) items. This is fine because pub-restricted item are not exported to other crates (so they are effectively private).